core/src/main/java/org/apache/calcite/plan/volcano/VolcanoPlanner.java

```diff
@@ -764,6 +764,13 @@ RelNode changeTraitsUsingConverters(
     // will be left as is.  Finally, any null entries in toTraits are
     // ignored.
     RelNode converted = rel;
+    Convention conventionTrait = null;
+    for (RelTrait trait : toTraits) {
+      if (trait instanceof  Convention) {
+        conventionTrait = (Convention) trait;
+        break;
+      }
+    }
```

Comment:
Can we use toTraits.getTrait() to get convention?

Reply:
yes, good idea.